### ISSUE TITLE
ec2_ami wait not long enough

### ISSUE THREAD

**Issue by dpeterka** [ansible/ansible#4619]
I randomly receive an error when waiting for ami creation, seems the call times out after a bit, even though the AMI create call is successful.

```
   - name: create ami
     action: ec2_ami
        instance_id=$ansible_ec2_instance_id
        wait=yes
        name={{ami_name}}
        state=present
     register: instance
```

File "/home/ec2-user/.ansible/tmp/ansible-1382371655.27-165243934146759/ec2_ami", line 1255, in <module>
    main()
  File "/home/ec2-user/.ansible/tmp/ansible-1382371655.27-165243934146759/ec2_ami", line 320, in main
    create_image(module, ec2)
  File "/home/ec2-user/.ansible/tmp/ansible-1382371655.27-165243934146759/ec2_ami", line 199, in create_image
    img = ec2.get_image(image_id)
  File "/usr/lib/python2.6/site-packages/boto/ec2/connection.py", line 256, in get_image
    return self.get_all_images(image_ids=[image_id], dry_run=dry_run)[0]
  File "/usr/lib/python2.6/site-packages/boto/ec2/connection.py", line 182, in get_all_images
    [('item', Image)], verb='POST')
  File "/usr/lib/python2.6/site-packages/boto/connection.py", line 1082, in get_list
    raise self.ResponseError(response.status, response.reason, body)
boto.exception.EC2ResponseError: EC2ResponseError: 400 Bad Request
<?xml version="1.0" encoding="UTF-8"?>
<Response><Errors><Error><Code>InvalidAMIID.NotFound</Code><Message>The image id '[ami-8271e8b2]' does not exist</Message></Error></Errors><RequestID>ede7c45e-fb8a-4

**Comment by mpdehaan** [comment]
What Ansible version did you experience this on?

(ansible --version?)

some changes to wait logic have been made on 1.4 (devel branch)

Thanks!

**Comment by dpeterka** [comment]
ansible --version
ansible 1.4 (devel 214b0b052c) last updated 2013/10/22 16:08:14 (GMT -400)

**Comment by jimi-c** [comment]
Following up on this. There is a `wait_timeout` parameter that is available, have you tried increasing that value? The default is 300 seconds.

**Comment by jimi-c** [comment]
After looking into the code, I see the portion in the traceback is not using the wait_timeout logic, so I've created the above patch to correct that. It will also now return an error message instead of failing with that traceback. Please let me know if you still continue having issues with this. Thanks!

**Comment by sayap** [comment]
I believe this was already fixed in a8f95435. Prior to a8f95435, line 199 is `img = ec2.get_image(image_id)`. In 214b0b0, it has already been moved to line 204, so the error that dpeterka encountered didn't come from 214b0b0.

In my opinion, d6b912c makes the code a bit more confusing by using the same `wait_timeout` to deal with EC2's eventual consistency model. We should always have some loop to deal with eventual consistency. Whether we want to wait until the image is available, that's another matter controlled by `wait_timeout`.
